### PR TITLE
Fix: Strip HTML tags from title (fixes #75)

### LINKED ISSUE #75: Theme styles can bleed into Course Map
### Subject of the issue
Theme styles can bleed into the Course Map if they're used in the display title. For example, this `displayTitle` contains `<span>` tags with theme classes, and those carry over into the Course Map.

```
"displayTitle": "Welcome to <span class='splash-title__course-name'>Client<i>Name's</i><sup>&trade;</sup> Training</span>",
```
Screenshot 2023-10-10 at 10 50 46 AM

### PR DESCRIPTION
**Pull request by swashbuck**
Fix #75 

### Fix
* Strips HTML from title text

### Testing
For the `displayTitle` of a component, use the following text:
```
Welcome to <span style='font-size: 3rem; color: red;'>Client<i>Name's</i><sup>&trade;</sup> Training</span>
```
The title should display like titles that do not contain HTML.

### CODE REVIEW

File: js/map.js [hunk 1]
```diff
@@ -246,8 +246,9 @@ class CourseMap extends Backbone.Controller {
       let t = this.get('displayTitle');
       if (isStringEmpty(t)) t = this.get('title');
       if (isStringEmpty(t)) t = this.get('_id');
+      
       // Strip HTML tags
-      let title = document.createElement("div");
+      const title = document.createElement("div");
```
Comment (oliverfoster) [inline js/map.js:251]: I think jquery has a function for this:

`$(t).text()`

https://api.jquery.com/text/

Comment (swashbuck) [reply js/map.js]: Thanks, that works. I still have to create the element so that jquery can retrieve the text value.

Comment (oliverfoster) [reply js/map.js]: Sorry, I forgot the wrapping

File: js/map.js [hunk 2]
```diff
@@ -246,6 +246,12 @@ class CourseMap extends Backbone.Controller {
       let t = this.get('displayTitle');
       if (isStringEmpty(t)) t = this.get('title');
       if (isStringEmpty(t)) t = this.get('_id');
+      
+      // Strip HTML tags
+      const title = document.createElement('div');
+      title.innerHTML = t;
+      t = $(title).text();
```
[the remark this answers is not in the document]
Comment (oliverfoster) [reply js/map.js]: There's some interesting distinction between a11y.normalize and a11y.removeBreaks
https://github.com/adaptlearning/adapt-contrib-core/blob/99d50626c2dc992b389a70f5ed5a1bec9af6f388/js/a11y.js#L798-L849

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**